### ISSUE TITLE
App not loading when accessed without trailing /

### ISSUE THREAD

**Issue by memen45** [owncloud/music#765]
When the app is accessed by the url "https://example.nextcloud.com/apps/music", the app keeps loading without finishing and the following appears in the console:

> Error: "Response for getList SHOULD be an array and not an object or something else"
>     Angular 7
> Possibly unhandled rejection: {}

Also, the url is now showing "https://example.nextcloud.com/apps/music#/".

When the app is accessed with the first url, but including a trailing "/", everything works fine. All other apps in nextcloud work fine either with or without trailing "/".

**Comment by memen45** [comment]
This is where the error comes from:

https://github.com/owncloud/music/blob/068c5f8b52679bab34dc587ee3c69d48ace1b041/js/vendor/restangular/dist/restangular.js#L1152

I checked the network tab and I see a problem here:

Without trailing "/":
> apps/api/prepare_collection
With trailing "/":
> apps/music/api/prepare_collection

It seems that the request url is not created correctly and the part after the last "/" is omitted and then appended with api/prepare_collection.

**Comment by paulijar** [comment]
Thanks for digging. I did notice earlier that the API calls were malformed. But what I didn't realize was that this was in fact the only problem. I thought that the routing of the AngularJS page would be broken when the address looked like "https://example.nextcloud.com/apps/music#/", and I tried to use redirection to make it look like "https://example.nextcloud.com/apps/music/#/". But this was a false assumption, in fact it doesn't matter that the address looks funny. The only important thing is that the base address of the RESTAngular library has to be set properly, to enable correctly formed API calls.

I still need to work out some details on how to set the base address so that it certainly works on all kinds of server configurations. But this shouldn't be a major obstacle, so the issue should get fixed for the next Music app release.
